producer.go

```diff
@@ -535,16 +535,25 @@ func (p *producer) dispatchWork(workCtx context.Context, count int, fetchResultC
 func (p *producer) heartbeatLogLoop(ctx context.Context) {
 	ticker := time.NewTicker(5 * time.Second)
 	defer ticker.Stop()
+	type state struct {
+		jobsRan    uint64
+		jobsActive int
+	}
+	prev := state{}
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			p.Logger.InfoContext(ctx, p.Name+": Heartbeat",
-				slog.Uint64("num_completed_jobs", p.numJobsRan.Load()),
-				slog.Int("num_jobs_running", int(p.numJobsActive.Load())),
-				slog.String("queue", p.config.Queue),
-			)
+			cur := state{jobsRan: p.numJobsRan.Load(), jobsActive: int(p.numJobsActive.Load())}
+			if cur != prev {
+				p.Logger.InfoContext(ctx, p.Name+": Heartbeat",
```

Comment:
Removing the guaranteed tick means this isn't really a heartbeat anymore. Maybe rephrase to?

```suggestion
				p.Logger.InfoContext(ctx, p.Name+": Producer job stats",
```

Reply:
Changed to `p.Logger.InfoContext(ctx, p.Name+": producer job state changed"`